File: AmeliorationVitesseProduction.cs
namespace exercice_trombone
{
    /// <summary>
    /// Effet d'amélioration de la vitesse de production des trombones.
    /// </summary>
    internal class AmeliorationVitesseProduction : EffetAmelioration
    {
        /// <inheritdoc/>
        public override void AppliquerEffet(Usine usine)
        {
            usine.VitesseProduction += valeur;
        }
    }
}

File: src/edu/cegepvicto/usine/AmeliorationVitesseProduction.java
package edu.cegepvicto.usine;

import java.lang.annotation.Inherited;

/**
 * Améliore la vitesse de production d'un usine.
 */
public class AmeliorationVitesseProduction extends EffetAmelioration {

    /**
     * @inheritDoc
     */
    @Override
    public void appliquerEffet(Usine usine) {
        usine.setVitesseProduction(usine.getVitesseProduction() + valeur);
    }
}
